File: SudokuCollective.WebApi/Models/Pagination/PageListModel.cs
using SudokuCollective.WebApi.Models.Enums;

namespace SudokuCollective.WebApi.Models.Pagination {

    public class PageListModel {

        public int Page { get; set; }
        public int ItemsPerPage { get; set; }
        public SortValue SortBy { get; set; }
        public bool OrderByDescending { get; set; }
        public bool IncludeCompletedGames { get; set; }

        public PageListModel () {

            Page = 0;
            ItemsPerPage = 0;
            SortBy = SortValue.NULL;
            OrderByDescending = false;
            IncludeCompletedGames = false;
        }

        public PageListModel (
            int page, 
            int itemsPerPage, 
            int sortValue, 
            bool orderByDescending,
            bool includeCompletedGames) {

            Page = page;
            ItemsPerPage = itemsPerPage;
            SortBy = (SortValue)sortValue;
            OrderByDescending = orderByDescending;
            IncludeCompletedGames = includeCompletedGames;
        }
    }
}

File: SudokuCollective.Api/Client/src/models/viewModels/pageListModel.js
import { SortValues } from "../viewModels/sortValues";

class PageListModel {

    constructor(page, itemsPerPage, sortBy, orderByDescending, includeCompletedGames) {

        if (!arguments.length) {

            this.page = 0;
            this.itemsPerPage = 0;
            this.sortBy = SortValues[0].value;
            this.orderByDescending = false;
            this.includeCompletedGames = false;

        } else {

            this.page = page;
            this.itemsPerPage = itemsPerPage;
            this.sortBy = sortBy;
            this.orderByDescending = orderByDescending;
            this.includeCompletedGames = includeCompletedGames;
        }
    }
}

export default PageListModel;